p2p/src/net/mock/mod.rs

```diff
@@ -87,26 +101,26 @@ impl NetworkService for MockService {
             return Err(P2pError::SocketError(ErrorKind::AddrNotAvailable));
         }
 
-        Ok((
-            addr,
-            MockSocket {
-                socket: TcpStream::connect(addr).await?,
-            },
-        ))
+        let (tx, rx) = oneshot::channel();
+        self.cmd_tx.send(types::Command::Connect { addr, response: tx }).await?;
+
+        let socket = rx
+            .await
+            .map_err(|e| e)? // channel closed
+            .map_err(|e| e)?; // command failure
```

Comment:
I don't get the `map_err` here. Seems like mapping an identity function should not do anything. Is there some funny thing with type casting going on?

Reply:
I prefer this over `let socket = rx.await??;` because it documents the two different errors that can happen when `rx` is polled.